packages/commonjs/src/transform-commonjs.js

```diff
@@ -94,6 +102,19 @@ export default function transformCommonjs(
         return;
       }
 
+      if (node.type === 'TryStatement') {
```

Comment:
I wonder if we could clean this up by moving most of this code into the "switch" statement?
For once, you can make use of the fact that for a `TryStatement` node, the `block` property is guaranteed to be present and to contain the corresponding BlockStatement: https://github.com/estree/estree/blob/master/es5.md#trystatement

So it should be possible to look for `TryStatement`s inside a new `case 'TryStatement':` below. Then you could set just the `tryCatchState.end` directly there to the end of the new TryStatement.

Going further, it seems you are using the depth to track nested try statements. I do not think this is necessary—I believe the feature would be simpler if you just ignore special handling for nested TryStatements, after all the code is already in the outer TryStatement?

Then you only need a single variable `currentTryBlockEnd: number | null` that serves the double purpose of telling us whether we are inside a try block as well as telling us where it ends.
- By default, it is `null`
- If the value is not null but `node.start > currentTryStatementEnd` we set it to `null`: We are just outside the try block
- In the switch statement `case 'TryStatement':`
  - You set this variable to `node.block.end` if it is `null`
  - Do nothing if it is already set—we ignore the nested statement

And the logic should become much smaller and not take up that much space and cognitive load in the general section of this function.

Reply:
We are *not* interested in the "end" of the `try` statement. We are waiting for the following block node to retrieve its end. As the "end" of the `try` statement includes the `catch` and `finally` stage.

Reply:
But now I do see that the node has a `block` property which could save us a few lines of code! :-)

Reply:
Yeah I got what you mean now lol